skimage/measure/_marching_cubes_lewiner_cy.pyx

```diff
@@ -0,0 +1,1463 @@
+            
+            TILING1, TILING2, TILING3_1, TILING3_2, TILING4_1, TILING4_2, 
+            TILING5, TILING6_1_1, TILING6_1_2, TILING6_2, TILING7_1, TILING7_2, 
+            TILING7_3, TILING7_4_1, TILING7_4_2, TILING8, TILING9, 
+            TILING10_1_1, TILING10_1_1_, TILING10_1_2, TILING10_2, TILING10_2_, 
+            TILING11, TILING12_1_1, TILING12_1_1_, TILING12_1_2, TILING12_2, 
+            TILING12_2_, TILING13_1, TILING13_1_, TILING13_2, TILING13_2_, 
+            TILING13_3, TILING13_3_, TILING13_4, TILING13_5_1, TILING13_5_2, 
+            TILING14,
+            
+            TEST3, TEST4, TEST6, TEST7, TEST10, TEST12, TEST13,
+            SUBCONFIG13,
+            ):
+        
+        self.EDGESRELX = Lut(EDGESRELX)
+        self.EDGESRELY = Lut(EDGESRELY)
+        self.EDGESRELZ = Lut(EDGESRELZ)
+        
+        self.CASESCLASSIC = Lut(CASESCLASSIC)
+        self.CASES = Lut(CASES)
+        
+        self.TILING1 = Lut(TILING1)
+        self.TILING2 = Lut(TILING2)
+        self.TILING3_1 = Lut(TILING3_1)
+        self.TILING3_2 = Lut(TILING3_2)
+        self.TILING4_1 = Lut(TILING4_1)
+        self.TILING4_2 = Lut(TILING4_2)
+        self.TILING5 = Lut(TILING5)
+        self.TILING6_1_1 = Lut(TILING6_1_1)
+        self.TILING6_1_2 = Lut(TILING6_1_2)
+        self.TILING6_2 = Lut(TILING6_2)
+        self.TILING7_1 = Lut(TILING7_1)
+        self.TILING7_2 = Lut(TILING7_2)
+        self.TILING7_3 = Lut(TILING7_3)
+        self.TILING7_4_1 = Lut(TILING7_4_1)
+        self.TILING7_4_2 = Lut(TILING7_4_2)
+        self.TILING8 = Lut(TILING8)
+        self.TILING9 = Lut(TILING9)
+        self.TILING10_1_1 = Lut(TILING10_1_1)
+        self.TILING10_1_1_ = Lut(TILING10_1_1_)
+        self.TILING10_1_2 = Lut(TILING10_1_2)
+        self.TILING10_2 = Lut(TILING10_2)
+        self.TILING10_2_ = Lut(TILING10_2_)
+        self.TILING11 = Lut(TILING11)
+        self.TILING12_1_1 = Lut(TILING12_1_1)
+        self.TILING12_1_1_ = Lut(TILING12_1_1_)
+        self.TILING12_1_2 = Lut(TILING12_1_2)
+        self.TILING12_2 = Lut(TILING12_2)
+        self.TILING12_2_ = Lut(TILING12_2_)
+        self.TILING13_1 = Lut(TILING13_1)
+        self.TILING13_1_ = Lut(TILING13_1_)
+        self.TILING13_2 = Lut(TILING13_2)
+        self.TILING13_2_ = Lut(TILING13_2_)
+        self.TILING13_3 = Lut(TILING13_3)
+        self.TILING13_3_ = Lut(TILING13_3_)
+        self.TILING13_4 = Lut(TILING13_4)
+        self.TILING13_5_1 = Lut(TILING13_5_1)
+        self.TILING13_5_2 = Lut(TILING13_5_2)
+        self.TILING14 = Lut(TILING14)
+        
+        self.TEST3 = Lut(TEST3)
+        self.TEST4 = Lut(TEST4)
+        self.TEST6 = Lut(TEST6)
+        self.TEST7 = Lut(TEST7)
+        self.TEST10 = Lut(TEST10)
+        self.TEST12 = Lut(TEST12)
+        self.TEST13 = Lut(TEST13)
+        
+        self.SUBCONFIG13 = Lut(SUBCONFIG13)
+
+
+def marching_cubes(im, double isovalue, LutProvider luts, int st=1, int classic=0):
```

Comment:
We can type `im` and even make it a memoryview right here in the definition. We've already run `volume = np.ascontiguousarray(volume, np.float32)` back in the Python wrapper, so the type and the C-ordering is guaranteed in memory.